common/src/main/java/com/wynntils/features/trademarket/TradeMarketQuickSearchFeature.java

```diff
@@ -0,0 +1,183 @@
+    // If we are searching such a potion, we simply search the second part. '.' matches the icon and '\w+' the skill.
+    // e.g. "Potion of ✤ Strength [2/2]" -> "Potion of ✤ Strength" -> "✤ Strength"
+    private static final Pattern POTION_PATTERN = Pattern.compile("^Potion of (. \\w+)");
+
+    // 'À' this char is behind all the items names and needs to be trimmed before we search.
+    // We cannot use WynnUtils.normalizeBadString() as this messes up dungeon keys.
+    // e.g. "CorruptedÀÀÀGalleon'sÀÀÀGraveyard KeyÀ" -> "CorruptedÀÀÀGalleon'sÀÀÀGraveyard Key"
+    private static final Pattern EOL_PATTERN = Pattern.compile("À+$");
+
+    private static final int SEARCH_SLOT = 47;
+    private String searchQuery;
+    private boolean inTradeMarket = false;
+    private boolean inSearchChat = false;
+    private boolean openChatWhenContainerClosed = false;
+    private boolean quickSearching = false;
+    private boolean instantSearching = false;
+
+    @SubscribeEvent
+    public void onScreenOpen(ScreenOpenedEvent.Post event) {
+        WynntilsMod.info("Screen Opened");
+        openChatWhenContainerClosed = false;
+        if ((Models.Container.getCurrentContainer() instanceof TradeMarketContainer)) {
+            inTradeMarket = true;
+            quickSearching = false;
+            inSearchChat = false;
+            return;
+        }
+        inTradeMarket = false;
+    }
+
+    @SubscribeEvent
+    public void onScreenClosed(ScreenClosedEvent event) {
+        if (inSearchChat && event.getScreen() instanceof ChatScreen) {
+            if (autoCancel.get() && KeyboardUtils.isKeyDown(GLFW.GLFW_KEY_ESCAPE)) {
+                McUtils.sendChat("cancel");
+            }
+            inSearchChat = false;
+        } else if (openChatWhenContainerClosed) {
+            if (quickSearching) {
+                McUtils.mc().setScreen(new ChatScreen(searchQuery));
+                quickSearching = false;
+            } else {
+                McUtils.mc().setScreen(new ChatScreen(""));
+            }
+            openChatWhenContainerClosed = false;
+            inSearchChat = true;
+        }
+    }
+
+    // EventPriority.HIGH so that InventoryEmeraldCountFeature does not render.
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    public void onMenuClosed(MenuClosedEvent event) {
+        WynntilsMod.info("Menu Closed");
+        if (!inTradeMarket) return;
+        inTradeMarket = false;
+        if (instantSearching) {
+            event.setCanceled(true);
+            instantSearching = false;
+        }
+    }
+
+    @SubscribeEvent
+    public void onChatMessageReceive(ChatMessageReceivedEvent event) {
+        if (!Models.WorldState.onWorld()) return;
+        StyledText styledText =
+                StyledTextUtils.unwrap(event.getOriginalStyledText()).stripAlignment();
+        if (!styledText.matches(TYPE_TO_CHAT_PATTERN)) return;
+        if (instantSearching) {
```

Comment:
Please leave a blank line after the set of checks for early return, not just here but everywhere in the code.

Reply:
The methods should be divided in two parts, first check if we really should be here, otherwise return, then if everything passes, do what we should do.

Reply:
[651e713](https://github.com/Wynntils/Wynntils/pull/2908/commits/651e713613b0c0675869651890dd5a0aab82988b)